hamilton/src/hybrid_planner.cpp

```diff
@@ -0,0 +1,576 @@
+			exit(-1);
+			}
+
+			succeeded = sparse_planner_.getPath(descartes_traj_output_);
+
+			if(!succeeded || descartes_traj_output_.empty())
+			{
+			ROS_ERROR_STREAM("Failed to retrieve robot path");
+			}
+		}
+		
+		ROS_INFO_STREAM("Task '"<<__FUNCTION__<<"' completed");
+	}
+
+	void HybridPlanner::runPath()
+	{
+		moveit_group_ptr_->setPlannerId(PLANNER_ID);//TOCHECK should be in init(). Also the config_ param or not doubt. Check header file. 
+		overall_robot_traj_.getRobotTrajectoryMsg(moveit_robot_traj_);
+
+		// creating goal joint pose to start of the path
+		// std::vector<double> seed_pose(robot_model_ptr_->getDOF());
+  		std::vector<double> start_pose;
+  		descartes_core::TrajectoryPtPtr first_point_ptr = path[0];
+		first_point_ptr->getNominalJointPose(config_.seed_pose,*descartes_robot_model_ptr_,start_pose);
+
+		// moving arm to joint goal
+		moveit_group_ptr_->setJointValueTarget(start_pose);
+		moveit_group_ptr_->setPlanningTime(config_.plan_time);
+		moveit_msgs::MoveItErrorCodes result = moveit_group_ptr_->move();
+		if(result.val != result.SUCCESS)
+		{
+		ROS_ERROR_STREAM("Move to start joint pose failed");
+		exit(-1);
+		}
+
+		// creating Moveit! joint trajectory from Descartes Trajectory
+		fromDescartesToMoveitTrajectory();
+
+		//all free space segments now in moveit_robot_traj_ and 
+		//all process segments in trajectory_msgs::JointTrajectory& joint_traj_descartes_
+		//TOCHECK TODO to combine them, again use robot_trajectory?
+	}
+
+	void HybridPlanner::fromDescartesToMoveitTrajectory()
+	{
+	  // Fill out information about our trajectory
+	  joint_traj_descartes_.header.stamp = ros::Time::now();
+	  joint_traj_descartes_.header.frame_id = config_.world_frame;
+	  joint_traj_descartes_.joint_names = config_.joint_names;
+
+	  // For keeping track of time-so-far in the trajectory
+	  double time_offset = 0.0;
+
+	  // Loop through the trajectory
+	  for (unsigned int i = 0; i < descartes_traj_input_.size(); i++)
+	  {
+	    // Find nominal joint solution at this point
+	    std::vector<double> joints;
+
+	    // getting joint position at current point
+	    const descartes_core::TrajectoryPtPtr& joint_point = descartes_traj_input_[i];
+	    joint_point->getNominalJointPose(std::vector<double>(), *descartes_robot_model_ptr_, joints);
+
+	    // Fill out a ROS trajectory point
+	    trajectory_msgs::JointTrajectoryPoint pt;
+	    pt.positions = joints;
+	    // velocity, acceleration, and effort are given dummy values
+	    // we'll let the controller figure them out
+	    pt.velocities.resize(joints.size(), 0.0);
+	    pt.accelerations.resize(joints.size(), 0.0);
+	    pt.effort.resize(joints.size(), 0.0);
+	    // set the time into the trajectory
+	    pt.time_from_start = ros::Duration(time_offset);
+	    // increment time
+	    time_offset += config_.time_delay;
+
+	    joint_traj_descartes_.points.push_back(pt);
+	  }
+
+	}
+}
+
```

Comment:
@jrgnicho  These were the misplaced comments

- Is that right?
To check: Is time being maintained correctly for each point

- Please have a look at void HybridPlanner::fromDescartesToMoveitTrajectory()
https://github.com/madratman/hamilton/blob/indigo-devel/hamilton/src/hybrid_planner.cpp#L500-537

- This v/s this:
```
typedef std::vector<descartes_core::TrajectoryPtPtr>::const_iterator JointSolutionIterator;
// For calculating the time_from_start field of the trajectoryPoint
ros::Duration time_from_start(time_offset);
std::vector<double> dummy_seed (robot_model.getDOF(), 0.0);
```
where time_offset is an argument.

- References for the two approaches:

https://github.com/ros-industrial-consortium/godel/blob/hydro-devel/godel_path_planning/src/trajectory_planning.cpp#L88-121

https://github.com/ros-industrial/industrial_training/blob/indigo-devel/training/ref/demo_descartes/src/plan_and_run/src/demo_application.cpp#L203-238

https://github.com/ros-industrial-consortium/descartes_tutorials/blob/master/descartes_tutorial1/src/tutorial1.cpp#L142-180